sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -293,14 +297,17 @@ class Analyzer(
       case p: LogicalPlan => p.transformExpressionsUp {
         case a @ Add(l, r) if a.childrenResolved => (l.dataType, r.dataType) match {
           case (CalendarIntervalType, CalendarIntervalType) => a
+          case (DateType, CalendarIntervalType) if conf.ansiEnabled => ANSIDateAdd(l, r)
```

Comment:
Can we just create a `DateAdd` expression, which has a `ansi` flag?

Reply:
Fixed